app/app.go

```diff
@@ -428,12 +429,17 @@ func NewOsmosisApp(
 		}
 
 		// Create the indexer streaming service.
+		rpcAddress, ok := appOpts.Get(rpcAddressConfigName).(string)
+		if !ok {
+			panic(fmt.Sprintf("failed to retrieve %s from config.toml", rpcAddressConfigName))
+		}
+		nodeStatusChecker := commonservice.NewNodeStatusChecker(rpcAddress)
```

Comment:
We already do this for sqs. Can we reduce code duplication and consolidate this initialization under a single execution flow?